canvasapi/outcome.py

```diff
@@ -0,0 +1,319 @@
+        """
+        List linked outcomes.
+
+        :calls: `GET /api/v1/global/outcome_groups/:id/outcomes \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.outcomes`_
+            or `GET /api/v1/accounts/:account_id/outcome_groups/:id/outcomes \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.outcomes`_
+            or `GET /api/v1/courses/:course_id/outcome_groups/:id/outcomes \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.outcomes`_
+
+        :returns: Paginated List of Outcomes linked to the group.
+        :rtype: :class:`canvasapi.paginated_list.PaginatedList` of
+            :class:`canvasapi.outcome.OutcomeLink`
+        """
+        return PaginatedList(
+            OutcomeLink,
+            self._requester,
+            'GET',
+            '%s/outcome_groups/%s/outcomes' % (self.context_ref(), self.id),
+            _kwargs=combine_kwargs(**kwargs)
+        )
+
+    def link_existing(self, outcome):
+        """
+        Link to an existing Outcome.
+
+        :calls: `PUT /api/v1/global/outcome_groups/:id/outcomes/:outcome_id \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.link`_
+            or `PUT /api/v1/accounts/:account_id/outcome_groups/:id/outcomes/:outcome_id \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.link`_
+            or `PUT /api/v1/courses/:course_id/outcome_groups/:id/outcomes/:outcome_id \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.link`_
+
+        :returns: OutcomeLink object with current OutcomeGroup and newly linked Outcome.
+        :rtype: :class:`canvasapi.outcome.OutcomeLink`
+        """
+        outcome_id = obj_or_id(outcome, "outcome", (Outcome,))
+
+        response = self._requester.request(
+            'PUT',
+            '%s/outcome_groups/%s/outcomes/%s' % (
+                self.context_ref(),
+                self.id,
+                outcome_id
+            )
+        )
+
+        return OutcomeLink(self._requester, response.json())
+
+    def link_new(self, title, **kwargs):
+        """
+        Create a new Outcome and link it to this OutcomeGroup
+
+        :calls: `POST /api/v1/global/outcome_groups/:id/outcomes \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.link`_
+            or `POST /api/v1/accounts/:account_id/outcome_groups/:id/outcomes \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.link`_
+            or `POST /api/v1/courses/:course_id/outcome_groups/:id/outcomes \
+            <https://canvas.instructure.com/doc/api/outcome_groups.html#method.outcome_groups_api.link`_
+
+        :params: title, description, ratings, and mastery points.
```

Comment:
docstring required for `title` (`:param title:` and `:type title:`).
No need to mention the others here.